### PR TITLE
Fix bug with row count comparison for expect_columns_equivalent().

### PR DESCRIPTION
**Pull request by nvdbaranec**
Top level row counts were being compared incorrectly for the `_equivalent()` check.   No tests were affected by this, thankfully.  In addition, even though `_equal()` was correct, I made the same modification to the call (passing separate index lists for the two columns) to future-proof against internal changes to the comparison function.

### CODE REVIEW

**Review by vuule — APPROVED** [review]

**Review by ttnghia — APPROVED** [review]
If you can add `const` after `auto` then it's perfect to me.

File: cpp/tests/utilities/column_utilities.cu
Comment (hyperbolic2346) [inline cpp/tests/utilities/column_utilities.cu:760]: why was the row count comparison removed here?

Comment (ttnghia) [reply cpp/tests/utilities/column_utilities.cu:760]: That means the result should be "false" instead of throwing exception.

Comment (nvdbaranec) [reply cpp/tests/utilities/column_utilities.cu:760]: Correct.  It resulted in inconsistent behavior.

**Review by hyperbolic2346 — APPROVED** [review]

**Comment by nvdbaranec** [comment]
I'm out until Monday, so let's hold off on merging this until then :)

**Comment by nvdbaranec** [comment]
rerun tests

**Comment by nvdbaranec** [comment]
@gpucibot merge

**Pull request merged**
